### ISSUE TITLE
more erroneous printingMathML

### ISSUE THREAD

**Issue by Karrenbelt** [sympy/sympy#17212]
I am opening another issue, related to #17184

```
sympy.mathml(sympy.true)
>>> '<booleantrue/>'
sympy.mathml(sympy.false)
>>> '<booleanfalse/>'
```
whereas I would expect simply `'<true/>'` and `'<false/>'`

The same goes for logical operators. The mathml for sympy.And and sympy.Or is correct
```
sympy.mathml(sympy.And(x, y))
>>> '<apply><and/><ci>x</ci><ci>y</ci></apply>'
sympy.mathml(sympy.Or(x, y))
>>> '<apply><or/><ci>x</ci><ci>y</ci></apply>'
```

But the mathml for those that yielded a 'Not Iterable' error before, the mathml is incorrect:

```
sympy.mathml(sympy.Implies(x, y))
>>> '<implies><ci>x</ci><ci>y</ci></implies>'
sympy.mathml(sympy.Not(x))
>>> '<not><ci>x</ci></not>'
sympy.mathml(sympy.Xor(x, y))
>>> '<xor><ci>x</ci><ci>y</ci></xor>'
```

and what I believe to be correct:
```
>>> '<apply><implies/><ci>x</ci><ci>y</ci></apply>'
>>> '<apply><not/><ci>x</ci></apply>'
>>> '<apply><xor/><ci>x</ci><ci>y</ci></apply>'
```

That might also be related to the following error, though I decided not to dig this one out:

```
sympy.mathml(sympy.Piecewise( (0, x <- 1), (f, x <= 1), (g, True)))
>>> AttributeError: 'str' object has no attribute 'nodeType'
```

What's more, sympy functions are not properly translated to mathML

```
sympy.mathml(sympy.Function('f')(1/x))
>>> '<apply><f/><apply><power/><ci>x</ci><cn>-1</cn></apply></apply>'
```

and I believe it should be:
`>>> '<apply><ci>f</ci><apply><divide/><cn>1</cn><ci>x</ci></apply></apply>'`

and for sympy.Lambda:
```
sympy.mathml(sympy.Lambda(x, x + 1))
>>> '<lambda><tuple><ci>x</ci></tuple><apply><plus/><ci>x</ci><cn>1</cn></apply></lambda>'
```
expected:
`>>> '<lambda><bvar><ci>x</ci></bvar><apply><plus/><ci>x</ci><cn>1</cn></apply></lambda>'`

**Comment by oscargus** [comment]
Are you sure that you have updated to the latest master version of SymPy? 

These are the tests that are used and passes:
https://github.com/sympy/sympy/blob/8ca4a683d58ac1f61cfd2e4dacf7f58b9c0fefab/sympy/printing/tests/test_mathml.py#L225-L229

https://github.com/sympy/sympy/blob/8ca4a683d58ac1f61cfd2e4dacf7f58b9c0fefab/sympy/printing/tests/test_mathml.py#L540-L542

Piecewise, Lambda, Functions are not (yet) supported though. Power instead of division is by SymPy design, there exists no division object in SymPy. Hence, to still print it like there was one require a bit of extra work, although it should be possible to follow e.g. the MathML presentation printer or (probably better) the LaTeX printer.

The MathML printer can be found here: https://github.com/sympy/sympy/blob/master/sympy/printing/mathml.py

Basically, it is a question of adding methods `_print_Piecewise`, `_print_Lambda`, and `_print_Function` (and corresponding tests).

There seems to be support for division already, see  https://github.com/sympy/sympy/blob/8ca4a683d58ac1f61cfd2e4dacf7f58b9c0fefab/sympy/printing/mathml.py#L205-L213 so it may have to do with the lacking support for printing Function in this case.

**Comment by oscargus** [comment]
It seems like Function is already supported. But by a design choice, any undefined function returns its own name as tag. The lines to consider are: https://github.com/sympy/sympy/blob/8ca4a683d58ac1f61cfd2e4dacf7f58b9c0fefab/sympy/printing/mathml.py#L193-L195
